src/ArenaReplay.cpp

```diff
@@ -600,7 +633,19 @@ class ReplayGossip : public CreatureScript
             }
             case REPLAY_LIST_BY_PLAYERNAME:
             {
-                QueryResult result = CharacterDatabase.Query("SELECT id FROM character_arena_replays WHERE winnerPlayerNames LIKE '%" + std::string(code) + "%' OR loserPlayerNames LIKE '%" + std::string(code) + "%'");
+                QueryResult resultGUID = CharacterDatabase.Query("SELECT guid FROM characters WHERE LOWER(name) = LOWER('" + std::string(code) + "')");
```

Comment:
we could use the cache here `sCharacterCache->GetCharacterGuidByName(name)` instead of calling the DB